src/util.rs

```diff
@@ -808,14 +808,17 @@ impl Run {
                 let (dag, mut cache) = rvsdg.to_dag_encoding(true);
 
                 let schedule_steps = parallel_schedule();
-                assert_eq!(
-                    schedule_steps.len(),
-                    1,
-                    "Parallel schedule had multiple steps!"
-                );
+                if schedule_steps.len() != 1 {
+                    log::warn!("Parallel schedule had multiple steps! You may need to adjust the schedule to make eggcc tractable.");
+                }
 
-                let egglog =
-                    build_program(&dag, Some(&dag), &dag.fns(), &mut cache, &schedule_steps[0]);
+                let egglog = build_program(
+                    &dag,
+                    Some(&dag),
+                    &dag.fns(),
+                    &mut cache,
+                    &schedule_steps.join("\n"),
```

Comment:
I think this is a bad idea since it's not something eggcc does. Probably we should make this give you the last egglog file, or it should work with `stop-after-n-passes`

Reply:
Will address this in my next PR